driver/src/contracts/stablex_contract.rs

```diff
@@ -120,6 +125,14 @@ pub trait StableXContract {
         gas_price: U256,
         block_timeout: Option<usize>,
     ) -> Result<(), MethodError>;
+
+    fn past_events(
+        &self,
+    ) -> BoxFuture<'static, Result<Vec<Event<batch_exchange::Event>>, ExecutionError>>;
+
+    fn stream_events(
+        &self,
+    ) -> BoxStream<'static, Result<Event<batch_exchange::Event>, ExecutionError>>;
```

Comment:
@nlordell You changed this to return a lifetime tied to self. Why? The futures we get from ethcontract are `'static` after all so we might as well show that in the interface.

Reply:
The owned events definitely are, but the `query_past_events_paginated` was written with `async` and I wasn't sure if the generated `Future` would have lifetimes tied to self.

Reply:
But rust would prevent us from having dangling references, no?

Reply:
Yes, but as far as I know a future can still have a lifetime that is tied to one of its parameters (like self or some buffer it’s supposed to write to).

I just thought ‘static added some extra unneeded restrictions to the trait but in this case it really isn’t a problem